docs/kickstarters/aws-microservices/technical-overview.md

```diff
@@ -0,0 +1,184 @@
+
+# Modules
+
+This boilerplate contains a simple web application to place a customer order for a product
+
+Core Functionalities of the application include
+* Adding Orders
+* Listing Orders
+* Notifying End Users with Order Information
+
+# Foundation Layers
+
+Following Layers will be part of the system.
+
+* Authentication and Authorization
+* Web GUI - The Technology used should be replaceable 
+* Microservices - Backend services to process requests and handle business logics
+* Database
+
+## Authentication and Authorization
+
+- OAuth2 Authentication method should be used to authenticate users.
+- Authorization should be enabled for role based access control.
+
+# Microservices
+
+* Order Service
+  - This service handles the CRUD operation of the customer order.
+
+
+* Omni Channel Notification Service 
+  - This service handles the transmission of the order information notifications to the end-user.
+
+  
+## Create a Customer Order
+**[POST] /api/customer-order**
+
+This endpoint is defined for creating a customer order. Following are the responsibilities of the endpoint.
+
+* Validate the Order request
+* Call the Notification Service and email the supplier on the placed order
+* Create a new record in the database
+
+**Header**
+
+* Authorization : Bearer <auth-token>
+* Content-Type : application/json
+
+
+**Body**
+```JSON
+{
+    "Order No": "string",
+    "Customer": "string",
+    "Supplier": "string",
+    "Product": "string"
+}
+```
+
+**Responses**
+- [201 - created] type of response will be returned as a success output together with the created order item in the response body.
+- [400 - bad request] type of response will be returned for validation errors with the error message in the request body.
+- [500 - internal server error] type of response will be returned for internal servers errors with the exception.
+
+
+## List Customer Orders
+**[GET] /api/customer-order**
+
+This endpoint is defined to retrive all customer orders from the system. Following are the responsibilities of the endpoint.
+* Return all customer orders from the system sorted by created date
+
+**Header**
+
+* Authorization : Bearer <auth-token>
+* Content-Type : application/json
+
+**Body**
+
+```JSON
+[{
+  "Order No": "string",
+  "Customer": "string",
+  "Product": "string",
+  "Supplier": "string",
+  "createdDate": "timestamp",
+  "status": "string"
+},
+{
+  "Order No": "string",
+  "Customer": "string",
+  "Product": "string",
+  "Supplier": "string",
+  "createdDate": "timestamp",
+  "status": "string"
+}]
+```
+
+**Responses**
+- [200 - ok] type response will be returned as a success output together with the list of customer orders in the response body.
+- [204 - no content] type response will be returned if there are no customer order found in the system.
+- [500 - internal server error] type of response will be returned for internal servers errors with the exception.
+
+## Approve Order
+**[PATCH] /api/customer-order/{id}/approve**
```

Comment:
Shall we make the URIs Restful by using plural nouns?
Examples:
[GET] /api/customer-orders/
[PATCH] /api/customer-orders/{id}?action=approve